libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -460,8 +461,19 @@ void AP_DDS_Client::on_topic (uxrSession* uxr_session, uxrObjectId object_id, ui
         }
         uint32_t* count_ptr = (uint32_t*) args;
         (*count_ptr)++;
-        GCS_SEND_TEXT(MAV_SEVERITY_INFO,"Received geometry_msgs/TwistStamped");
-        // TODO implement the velocity control to AP
+        auto *external_control = AP::externalcontrol();
+        if (external_control == nullptr) {
+            break;
+        }
```

Comment:
Tested it with `ardupilot_gz`. I did a takeoff and put the copter into guided mode with a GCS.
I then used the following terminal command to send in velocity inputs through ros 2/dds:
```
ros2 topic pub /ap/cmd_vel geometry_msgs/msg/TwistStamped '{twist: {angular: {z: 1.0}}}'
ros2 topic pub /ap/cmd_vel geometry_msgs/msg/TwistStamped '{twist: {linear: {z: 1.0}}}'
```
In both cases it entered  this condition, resulting in the copter not moving.

Reply:
This was caused by the singleton, the class constructor that attributes it was missing. Me and Ryan had a quick debug meeting and fixed it [on his branch](https://github.com/Ryanf55/ardupilot/commit/8aed4220628dfc856d22deda008eead328da10cb)

Reply:
Fixed in 657d9da8372238907bdb10aa53314f5a1b413f50